cdap-ui/app/cdap/components/AbstractWidget/JoinTypeWidget/index.tsx

```diff
@@ -0,0 +1,200 @@
+import PropTypes from 'prop-types';
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+
+import Select from 'components/AbstractWidget/SelectWidget';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import Paper from '@material-ui/core/Paper';
+
+import ThemeWrapper from 'components/ThemeWrapper';
+import If from 'components/If';
+
+const styles = (theme) => {
+  return {
+    multiCheckboxesContainer: {
+      padding: '20px',
+    },
+    empty: {
+      backgroundColor: 'initial',
+      padding: 'initial',
+      '> h4': {
+        marginBottom: '0px',
+      },
+    },
+    checkboxesGroup: {
+      display: 'flex',
+      'flex-direction': 'column',
+    },
+    subtitle: {
+      color: '#3c4355',
+      fontSize: '13px',
+      'font-weight': '500',
+      marginTop: '15px',
+      marginBottom: '5px',
+    },
+    textWarning: {
+      marginBottom: '5px',
+    },
+    checkbox: {
+      width: '45%',
+      display: 'inline-flex',
+    },
+  };
+};
+
+interface IJoinTypeWidgetProps extends WithStyles<typeof styles> {
+  value: string;
+  inputSchema: Array<{ name: string }>;
+  onChange: (arg0: string) => void;
+}
+
+const DROP_DOWN_OPTIONS: string[] = ['Inner', 'Outer'];
+
+function JoinTypeWidgetView({ value, inputSchema, onChange, classes }) {
+  const [joinType, setJoinType] = useState(DROP_DOWN_OPTIONS[1]);
+  const [selectedCount, setSelectedCount] = useState(0);
+  const [inputs, setInputs] = useState([]);
+
+  const formatOutput = () => {
+    const outputArr = inputs.filter((schema) => schema.selected).map((schema) => schema.name);
+    onChange(outputArr.join(','));
+    setSelectedCount(outputArr.length);
+  };
+
+  const joinTypeChange = (event) => {
+    setJoinType(event.target.value);
+    setInputs(
+      inputSchema.map((input) => {
+        return { name: input.name, selected: event.target.value === 'Inner' };
+      })
+    );
+  };
+
+  const checkBoxChange = (event) => {
+    setInputs(
+      inputs.map((input) => {
+        if (input.key === event.target.value) {
+          input.selected = !input.selected;
+        }
+        return input;
+      })
+    );
+  };
+
+  useEffect(() => {
+    const initialModel = value.split(',').map((input) => input.trim());
+    if (!value) {
+      setInputs(
+        inputSchema.map((input, i) => {
+          return { name: input.name, key: `${i}-${input.name}`, selected: false };
+        })
+      );
+    }
+    if (initialModel.length === inputSchema.length) {
+      setJoinType('Inner');
+      setInputs(
+        inputSchema.map((input, i) => {
+          return { name: input.name, key: `${i}-${input.name}`, selected: true };
+        })
+      );
+    } else {
+      setJoinType('Outer');
+      setInputs(
+        inputSchema.map((input, i) => {
+          return {
+            name: input.name,
+            key: `${i}-${input.name}`,
```

Comment:
you already set the key on the actual render method. Do you still need the keys here?